hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/DecodingStrategies.java

```diff
@@ -65,6 +69,54 @@ TokenTransferList convert(
                 @NonNull AddressIdConverter addressIdConverter);
     }
 
```

Comment:
I image that this class and perhaps `HtsCallAttempt` will see a lot of traffic from different people when implementing all of the other calls.  I wonder if it makes sense to break these out into separate classes?  Not now but maybe something to worry about later...

Reply:
That's a good point...I think we should really have a `CallProvider<T extends HtsCall>` type, where each `HtsCallAttempt` is constructed with a reference to a `@Singleton` `List<CallTranslator<T extends HtsCall>> callTranslators` which it will just iterate through to find the necessary translation logic.

Will do now.